openapi/src/main/java/io/micronaut/openapi/visitor/OpenApiApplicationVisitor.java

```diff
@@ -772,12 +773,117 @@ private OpenAPI postProcessOpenApi(OpenAPI openApi, VisitorContext context) {
         }
 
         removeEmtpyComponents(openApi);
+        findAndRemoveDuplicates(openApi);
 
         openApi = resolvePropertyPlaceHolders(openApi, context);
 
         return openApi;
     }
 
+    private <T> List<T> findAndRemoveDuplicates(List<T> elements, BiPredicate<T, T> predicate) {
```

Comment:
would this not be better written with `elements.stream().distinct()` or `elements.stream().filter(..)`?

Reply:
@graemerocher Not for everyone... Personally, I try not to use streams because of their slowness and the creation of a bunch of additional objects. Yes, it is more convenient in terms of syntax, but always worse in performance